### PR TITLE
CI additions: now runs clang and doxygen and builds esp target using …

### PR DESCRIPTION
**Pull request by g-lesssard**
CI additions: now runs clang and Doxygen and builds esp target using a SwarmUs owned docker image.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ColumnLimit: 100
+BinPackParameters: false
+AlwaysBreakTemplateDeclarations: true
+PenaltyReturnTypeOnItsOwnLine: 10000
+BreakConstructorInitializers: AfterColon
+ConstructorInitializerAllOnOneLineOrOnePerLine: true
+PointerAlignment: Left
+
+# Clang version mismatch fixes
+AlignTrailingComments: false
```
Comment (xgroleau) [inline .clang-format:14]: Check out the update here. It allows us to match ou naming to the one we agreed on for C++
https://github.com/SwarmUS/HiveMind/blob/feat/logging/.clang-tidy

File: tools/run-clang-format.py [hunk 2]
```diff
@@ -0,0 +1,419 @@
+#!/usr/bin/env python
```
Comment (xgroleau) [inline tools/run-clang-format.py:1]: Heads up, you need to paste their license somewhere. Checkout https://github.com/SwarmUS/HiveMind/blob/feat/logging/THIRD-PARTY-NOTICES.md

File: .github/workflows/cmake.yml [hunk 3]
```diff
@@ -6,28 +6,46 @@ on:
...
 
-env:
-  CMAKE_TEST_PORJECT_PATH: build/tests
-  CMAKE_TEST_PORJECT_NAME: Template_CXX_tests
-
 jobs:
   build:
     runs-on: ubuntu-latest
+    container:
+      image: swarmus/github-ci-action
```
Comment (xgroleau) [inline .github/workflows/cmake.yml:13]: Maybe swarmus/esp-idf or swarmus/embedded, something like that would be more accurate? Maybe eventually we will have an image for the C# repo for AR.  I know there's all the dependencies with ROS and all, i'm just note sure of the name. Where is the docker file btw?

**Review by xgroleau — COMMENTED** [review]
GJ, just a couple comments that shouldn't be too long to fix

File: .clang-format (hunk 1 above)
Comment (g-lesssard) [reply .clang-format:14]: Added esp-idf license as well

File: tools/run-clang-format.py (hunk 2 above)
Comment (g-lesssard) [reply tools/run-clang-format.py:1]: Added esp-idf license as well

File: .github/workflows/cmake.yml (hunk 3 above)
Comment (g-lesssard) [reply .github/workflows/cmake.yml]: Switched to "embedded" image Charles made for ROS.

**Review by xgroleau — APPROVED** [review]

**Pull request merged**